src/lib.rs

```diff
@@ -269,3 +296,119 @@ impl<'a> AsRef<[u8]> for Diff<'a> {
         self.bytes.as_ref()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use std::{sync::Arc, time::Duration};
+
+    use iroh::protocol::Router;
+    use loro::{EventTriggerKind, LoroDoc};
+    use testresult::TestResult;
+    use tracing::{error_span, info, Instrument};
+    use tracing_test::traced_test;
+
+    use crate::{IrohLoroProtocol, SyncMode};
+
+    async fn setup_node(doc: LoroDoc) -> TestResult<(Router, IrohLoroProtocol)> {
+        let proto = IrohLoroProtocol::new(doc);
+        let endpoint = iroh::Endpoint::builder().bind().await?;
+
+        // Create and configure iroh node
+        let router = iroh::protocol::Router::builder(endpoint)
+            .accept(IrohLoroProtocol::ALPN, proto.clone())
+            .spawn()
+            .await?;
+        Ok((router, proto))
+    }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn basic() -> TestResult<()> {
```

Comment:
You can actually omit the `<()>`:
```suggestion
    async fn basic() -> TestResult {
```
But that's fine - that's now a super nitty nit :D